File: ui-admin/src/api/resource.ts
import request from './request'

export interface ResourceVO {
  id: string
  groupId: string
  resName: string
  resType: 'API' | 'BUTTON'
  resPath: string
  resCode: string
  resMethod: string
  description: string | null
  sort: number
  status: number  // 0=正常, 1=停用
  createTime: string
}

export interface ResourceQueryDTO {
  groupId?: string
  resName?: string
  resType?: string
  resCode?: string
  status?: number
  current?: number
  size?: number
}

export interface ResourceCreateDTO {
  groupId: string
  resName: string
  resType: 'API' | 'BUTTON'
  resPath: string
  resCode: string
  resMethod?: string
  description?: string
  sort?: number
}

export interface ResourceUpdateDTO {
  resName?: string
  resType?: 'API' | 'BUTTON'
  resPath?: string
  resMethod?: string
  description?: string
  sort?: number
  status?: number
}

export interface PageResult<T> {
  records: T[]
  total: number
  size: number
  current: number
}

/**
 * 分页查询资源列表
 */
export const getResourcePage = (params: ResourceQueryDTO): Promise<PageResult<ResourceVO>> => {
  return request.get<any, PageResult<ResourceVO>>('/system/resources', { params })
}

/**
 * 获取资源详情
 */
export const getResourceDetail = (id: string): Promise<ResourceVO> => {
  return request.get<any, ResourceVO>(`/system/resources/${id}`)
}

/**
 * 创建资源
 */
export const createResource = (data: ResourceCreateDTO): Promise<string> => {
  return request.post<any, string>('/system/resources', data)
}

/**
 * 更新资源
 */
export const updateResource = (id: string, data: ResourceUpdateDTO): Promise<void> => {
  return request.put<any, void>(`/system/resources/${id}`, data)
}

/**
 * 删除资源
 */
export const deleteResource = (id: string): Promise<void> => {
  return request.delete<any, void>(`/system/resources/${id}`)
}

/**
 * 批量删除资源
 */
export const deleteResourceBatch = (ids: string[]): Promise<void> => {
  return request.delete<any, void>('/system/resources/batch', { data: ids })
}

File: server/system/src/main/java/com/github/motoryang/system/modules/resource/entity/Resource.java
package com.github.motoryang.system.modules.resource.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.github.motoryang.common.core.domain.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 资源实体
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("sys_resource")
public class Resource extends BaseEntity {

    /**
     * 资源组ID
     */
    private String groupId;

    /**
     * 资源名称
     */
    private String resName;

    /**
     * 资源类型（API、BUTTON）
     */
    private String resType;

    /**
     * 资源路径（API路径或组件ID）
     */
    private String resPath;

    /**
     * 资源编码
     */
    private String resCode;

    /**
     * 请求方式（GET/POST/PUT/DELETE/*）
     */
    private String resMethod;

    /**
     * 资源描述
     */
    private String description;

    /**
     * 显示顺序
     */
    private Integer sort;

    /**
     * 状态（0正常 1停用）
     */
    private Integer status;
}
